app/lib/backend/model_extensions/service.rb

```diff
@@ -34,11 +33,9 @@ def update_backend_service
       end
 
       def delete_backend_service
-        if account && account.has_bought_cinstance? && !account.api_key.blank?
-          delete_alert_limits(alert_limits)
-          ThreeScale::Core::Service.delete_by_id!(backend_id)
-        end
-
+        return if account && !account.api_key?
```

Comment:
What I said at https://github.com/3scale/porta/pull/19#discussion_r263302277 was that if you also define

```ruby
class Account
  def missing_api_key?
    !api_key?
  end
end
```

then...

```ruby
return if account&missing_api_key?
```

Reply:
@guicassolato That wouldn't be correct, we want it to return if the api key is missing AND the account still exists.
`return if account&.missing_api_key?` would return if the account does not exist anymore, and that's wrong 😞 
So it would be `return if account && account.missing_api_key?` if I did that method, and I am not sure it is worth a method to avoid the `!` in here then 😜

Reply:
If the account does not exist anymore then it will be `return if nil`, no?

Reply:
@guicassolato As I told Michal in https://github.com/3scale/porta/pull/19#discussion_r263295956

> If we remove the tenant, then its service it is destroyed, and we want it to be destroyed from Backend too then 😄
> Actually that's the only reason why I changed this method, only to accomplish that 😄

If we return when the account does not exist anymore, as it was doing before, then the service will not be deleted from backend, and that does not make sense 😄

Reply:
I think `return if nil` means it will NOT return, @Martouta .

Reply:
ah right 🤦‍♀️ sorry, my bad 👍 Will do it 👍

Reply:
@guicassolato Done 💪 Thanks 💃